person.rb

```diff
@@ -13,9 +19,42 @@ def can_use_services?
     of_age? || @parent_permission
   end
 
-  private
+  def correct_name
+    @name
+  end
 
   def of_age?
     @age >= 18
   end
 end
+
+class BaseDecorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < BaseDecorator
+  def correct_name
+    name = @nameable.correct_name
+    name.length > 10 ? name[0..9] : name
+  end
+end
```

Comment:
- [ ] [OPTIONAL] Kindly move the `Nameable`, `BaseDecorator`, `CapitalizeDecorator`, and `TrimmerDecorator` classes to a separate file in order to make your code clean and professional.